exo-go/src/template_helpers/index.go

```diff
@@ -167,3 +168,49 @@ func CreateTmpServiceDir(chosenTemplate string) string {
 	}
 	return serviceTmpDir
 }
+
+// AddTemplate fetches a remote template from GitHub and stores it
+// under templateDir, returns an error if any
+func AddTemplate(gitURL, templateName, templateDir string) error {
+	if output, err := processHelpers.Run("", "git", "submodule", "add", "--name", templateName, gitURL, templateDir); err != nil {
```

Comment:
The first argument is the path that the command should be run in. This technically should be `cwd` but I didn't want to repeat the code below for every function that calls `Run`
```
cwd, err := os.Getwd()
if err != nil {
   log.Fatal("some message")
}
```